packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -64,6 +65,14 @@ export function containsDynamicChildren(children: ChildNode[]): boolean {
             return containsDynamicChildren(child.children);
         }
 
+        if (isIfBlock(child)) {
+            const trueConditionHasDynamicChildren = containsDynamicChildren(child.children);
+            const elseConditionHasDynamicChildren = child.else
+                ? containsDynamicChildren(child.children)
```

Comment:
Both of these are checking `containsDynamicChildren(child.children)`, so we're missing an entire branch of checks under the else condition. 

Did you mean to check the children of the elseifBlock/elseBlock here for `elseConditionHasDynamicChildren`?

```
const elseConditionHasDynamicChildren = child.else
    ? // recurse on child.else which could be ElseifBlock or ElseBlock
    : false;
```

Reply:
Good catch!  Yes, meant to recurse on the else branch.